lib/mutgen.c

```diff
@@ -25,6 +25,17 @@
 
 #include "msprime.h"
 
+typedef struct {
+    tsk_mutation_t *mutation;
+    struct mutation_node_t *next;
+} mutation_node_t;
```

Comment:
``mutation_list_t`` and ``site_list_t`` might be more intuitive, as "node" is something we use a lot (although "node" is technically more correct since these are nodes in the linked list)